tests/tests.lua

```diff
@@ -30,48 +30,31 @@ local machine_config = ".sunodo/image"
 local machine_runtime_config = { skip_root_hash_check = true }
 local machine_remote_protocol = "jsonrpc"
 
-local CodecOpcodes = {
-    BugLessState = 0x57896b8c,
-    CreateAppBounty = 0x5792993c,
-    AddSponsorship = 0x43032fa8,
-    WithdrawSponsorship = 0x7e9de47b,
-    SendExploit = 0xc2edf048,
-    TestExploit = 0x18cc70af,
-}
-
 --------------------------------------------------------------------------------
 -- Utilities
 
 local function decode_response_jsons(res)
     for _, v in ipairs(res.reports) do
-        local status = string.byte(v.payload:sub(1, 1))
-        if status == 0 then -- log
-            local text = v.payload:sub(2)
-            res.error = (res.error or "") .. text
-        elseif status == 1 then
-            local opcode = string.unpack(">I4", v.payload:sub(2, 5))
-            assert(opcode == CodecOpcodes.BugLessState, "unexpected bugless state")
-            local body = v.payload:sub(6)
-            res.state = fromjson(body)
-        else
-            error("unknown response status " .. status)
+        local body = v.payload
+        local ok, state = pcall(fromjson, body)
+        if ok then
+            res.state = state
```

Comment:
One difference between EggRoll and Rollmelette is that the back-end doesn't emit reports with error messages anymore. Now, the report is the JSON-encoded application state, with no extra bytes or anything. You'll notice I've also removed the assertions regarding `res.error`.